push.go

```diff
@@ -35,15 +34,33 @@ func (cmd *PushCommand) Run() error {
 		return err
 	}
 
-	formats, err := client.FormatsList(1, 25)
-	if err == nil {
-		err = sources.setFormats(formats)
-		if err != nil {
+	formatMap, _ := GetFormats(client)
+	for _, source := range sources {
+		formatName := source.GetFileFormat()
+		if val, ok := formatMap[formatName]; ok {
+			source.Format = val
+		}
+		if source.Format == nil {
+			return fmt.Errorf("Format %q of source %q is not supported by PhraseApp!", source.Format.ApiName, source.File)
 		}
 	}
 
+	projectIds := []string{}
```

Comment:
this looks quite like a duplication of this bit here https://github.com/phrase/phraseapp-client/pull/43/files#diff-27c9dec751c416fdf10e5e5e780b008cR35

Maybe `Targets` and `Sources` could just implement an interface `ProjectIDs() []string`?